std/os/linux/test.zig

```diff
@@ -42,3 +44,39 @@ test "timer" {
     // TODO implicit cast from *[N]T to [*]T
     err = linux.epoll_wait(@intCast(i32, epoll_fd), @ptrCast([*]linux.epoll_event, &events), 8, -1);
 }
+
+export fn iter_fn(info: *linux.dl_phdr_info, size: usize, data: ?*usize) i32 {
+    var counter = data.?;
+
+    // Count how many libraries are loaded
+    counter.* += usize(1);
+
+    const dlpi_name = info.dlpi_name orelse return -1;
+    // Make sure the name is not corrupted
+    if (!mem.eql(u8, mem.toSliceConst(u8, dlpi_name), "/proc/self/exe")) return -1;
+    // At least a PT_PHDR section and a PT_LOAD one
+    if (info.dlpi_phnum < 2) return -1;
+
+    // Quick & dirty validation of the phdr pointers, make sure we're not
+    // pointing to some random gibberish
+    var i: usize = 0;
+    while (i < info.dlpi_phnum) : (i += 1) {
+        const p_type = info.dlpi_phdr[i].p_type;
+        const valid = switch (info.dlpi_phdr[i].p_type) {
+            elf.PT_NULL ... elf.PT_NUM => true,
+            elf.PT_LOOS ... elf.PT_HIOS => true,
+            elf.PT_LOPROC ... elf.PT_HIPROC => true,
+            else => false,
+        };
+        if (!valid) return -1;
+    }
+
+    return 0;
+}
+
+test "dl_iterate_phdr" {
+    var counter: usize = 0;
+    const r = linux.dl_iterate_phdr(usize, iter_fn, &counter);
+    expect(r == 0);
+    expect(counter == 1);
```

Comment:
Is this liable to break in future if we somehow end up using other libraries in the test suite?

Reply:
Yep, but the test is easy to change when/if that happens.